charts/portworx/templates/storage-cluster.yaml

```diff
@@ -304,9 +304,16 @@ spec:
     {{- end }}
   {{- end }}
 
-  {{- if or (eq .Values.csi true) (eq .Values.csi false) }}
+  {{- if or (eq .Values.csi.enabled true) (eq .Values.csi.enabled false) }}
   csi:
-    enabled: {{ .Values.csi }}
+    enabled: {{ .Values.csi.enabled }}
+    {{- if eq .Values.csi.topology true}}
```

Comment:
should this be not .Values.csi.topology.enabled instead of .Values.csi.topology ?

https://github.com/libopenstorage/operator/blob/a57ea272d5c23dbb070994421a4dc06eef14b1b1/pkg/apis/core/v1/storagecluster.go#L199

Reply:
Since we have only on field in topology, using a single `topology` field keeps the configuration straightforward and user-friendly. It avoids unnecessary complexity and makes it easier for users to enable or disable the feature. 

For CSI top-level `csi.enabled` toggle makes it easy to enable or disable CSI without navigating through nested settings. It ensures that users can quickly manage the overall activation of CSI without complexity.

Reply:
Done, @nikita-bhatia  . I have added `topology.enabled` as per our discussion.